index.js

```diff
@@ -1,84 +1,173 @@
-                    send(message);
-                    console.log(message);
-                  }
-                }
+            timeout: 2000,
+            followRedirect: true
+        }, (error, response, body) => {
+            const responseDate = new Date();
+
+            if (error !== null) {
+                handleLocalError(name, responseDate);
+            } else if (response.statusCode == 200) {
+                handleSuccessStatus(name, url, responseDate);
             } else {
-                if (typeof notReachable[name] == "undefined") { // first time
-                    notReachable[name] = "true";
-                    var message = name + " " + emoji.get('hankey') + " " + getFormattedDate();
-                    console.log("first time :" + message);
-
-                } else if (notReachable[name] == "true") { // secondtime
-                    notReachable[name] = "notagain"; // we don't want second notification
-                    var message = name + " " + emoji.get('hankey') + " " + getFormattedDate();
-                    console.log(message);
-                    send(message);
-                }
+                handleDownStatus(name, url, responseDate, mentionsWhenDown);
             }
         });
     }, interval);
 }
 
+function handleSuccessStatus(urlName, urlValue, responseDate) {
+    const message = `:globe_with_meridians: ${urlName} (${urlValue}) is up and running!`;
+
+    if (urlContainer[urlName] === undefined) {
+        send("First check of this session... Endpoint is running! :beer:");
+    } else {
+        if (urlErrorContainer[urlName] !== undefined) {
+            restoreLocalErrorAndSend(responseDate, urlName);
+        }
+
+        if (urlContainer[urlName] !== UrlStatus.Up) {
+            send(message);
+        }
+    }
+
+    console.log(`${getFormattedDate(responseDate)} - ${message}`);
+    urlContainer[urlName] = UrlStatus.Up;
+}
+
+function handleLocalError(urlName, responseDate) {
+    console.log('Local error, couldn\'t reach endpoint.');
+
+    if (urlErrorContainer[urlName] === undefined) {
+        urlErrorContainer[urlName] = responseDate;
+    }
+}
+
+function handleDownStatus(urlName, urlValue, responseDate, mentionsWhenDown) {
+    const trailingMentions = buildMentionsForDownStatus(mentionsWhenDown);
+    const message = `:warning: Oh no! It looks like ${urlName} (${urlValue}) is down.\n${trailingMentions}`;
+    console.log(`${getFormattedDate(responseDate)} - ${message}`);
+
+    if (urlContainer[urlName] === undefined) {
+        send("First check of this session... Endpoint didn't respond :confused:");
+        urlContainer[urlName] = UrlStatus.FirstDown;
+        return;
+    }
+
+    if (urlErrorContainer[urlName] !== undefined) {
+        restoreLocalErrorAndSend(responseDate, urlName);
+    }
+
+    if (urlContainer[urlName] === UrlStatus.Up) {
+        // First fail! Try again, maybe this was temporary.
+        urlContainer[urlName] = UrlStatus.FirstDown;
```

Comment:
nit: you could simplify the statuses as `Up`, `Pending`, and `Down` and use some counter that is configurable, checking how many fail until considering down. Then, you'd only `send` message when is transitioning from `Pending/Checking` to `Down`

Reply:
I'd say that Kubernetes' probes are a good design on what to implement as parameters for a health check: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes